### ISSUE TITLE
Final scripts aliases fail due to trailing slash when `require`d

### ISSUE THREAD

**Issue by Alexis-ROYER** [evanw/esbuild#2730]
# Problem
When [alias](https://esbuild.github.io/api/#alias) is resolved with a final .js script (1), and a `require()` call adds a trailing slash after the package name (2), esbuild fails with the following error:
```
X [ERROR] Could not resolve "(...)\\node_modules\\util\\util.js/" (originally "util/")

    ../../node_modules/assert/assert.js:73:19:
      73 │ var util = require('util/');
         ╵                    ~~~~~~~

  The path "util/" was remapped to "(...)\\node_modules\\util\\util.js/" using the alias feature, which then couldn't be resolved. Keep in mind that import path aliases are resolved in the current working directory.

  You can mark the path "(...)\\node_modules\\util\\util.js/" as external to exclude it from the bundle, which will remove this error. You can also surround this "require" call with a try/catch block to handle this failure at run-time instead of bundle-time.
```

(1) As [node-libs-react-native@1.2.1](https://github.com/parshap/node-libs-react-native/blob/v1.2.1/index.js#L36) does.
(2) As [commonjs-assert@1.5.0](https://github.com/browserify/commonjs-assert/blob/v1.5.0/assert.js#L73) does.

# Example
Reuse of [node-libs-react-native](https://github.com/parshap/node-libs-react-native)'s polyfills in esbuild configuration:
```js
const nodeLibsReactNative = require("node-libs-react-native");
for (const _packageName in nodeLibsReactNative) {
    /** @const {string | null} */ const _packagePath = nodeLibsReactNative[_packageName];
    if (_packagePath) {
        esbuildOptions.alias[_packageName] = _packagePath;
    }
}
```
Environment: React Native, Android

# Workaround
Based on the previous configuration example, I had to fix [node-libs-react-native](https://github.com/parshap/node-libs-react-native) final script paths to package directory paths for a couple of modules:
```js
for (const _packageName in nodeLibsReactNative) {
    (...)
}
const path = require("path");
esbuildOptions.alias["util"] = path.resolve(__dirname, "../../node_modules/util/");
```

**Comment by Alexis-ROYER** [comment]
Not sure whether this is a bug or just a misuse of the feature.
Maybe it is nonsense to set final script paths with it?

**Comment by alxroyer** [comment]
Tested : fixed with esbuild@0.16.7.
